db/view/view.cc

```diff
@@ -518,18 +518,37 @@ row_marker view_updates::compute_row_marker(const clustering_or_static_row& base
      *      will all unselected columns.
      */
 
-    // WARNING: The code assumes that if multiple regular base columns are present in the view key,
-    // they share liveness information. It's true especially in the only case currently allowed by CQL,
-    // which assumes there's up to one non-pk column in the view key. It's also true in alternator,
-    // which does not carry TTL information.
+    // We need to pick one liveness (timestamp and expiration) for the row
+    // marker, which in general would be a problem if there are multiple
+    // regular base columns in the view key, as those can have different
+    // liveness. This is why we generally (in CQL) do NOT allow multiple
+    // regular base columns in the view key.
+    // We make an exception for Alternator, and allow two columns there (as
+    // the DynamoDB API requires us to allow). We can allow this because
+    // Alternator doesn't use expiration (the "TTL" it does support is
+    // different), and doesn't support user-defined timestamps. But, the
+    // two columns can still have different timestamps - this happens if an
+    // update modifies just one of them. The timestamp of the view update
+    // is the later of the two updated columns.
     const auto& col_ids = base_row.is_clustering_row()
             ? _base_info->base_regular_columns_in_view_pk()
             : _base_info->base_static_columns_in_view_pk();
     if (!col_ids.empty()) {
         auto& def = _base->column_at(base_row.column_kind(), col_ids[0]);
         // Note: multi-cell columns can't be part of the primary key.
         auto cell = base_row.cells().cell_at(col_ids[0]).as_atomic_cell(def);
-        return cell.is_live_and_has_ttl() ? row_marker(cell.timestamp(), cell.ttl(), cell.expiry()) : row_marker(cell.timestamp());
```

Comment:
to be alerted in case there are some more landmines in this area perhaps worth adding 2 logic_error throws:
- if there is more than 2 col_ids
- if there is 2 but any of them has ttl

Reply:
I can do that. It shouldn't happen (CQL doesn't give you a way to use more than 1, the DynamoDB API doesn't have a syntax for adding more than 2), but I guess it can't hurt to add an on_internal_error() or something.